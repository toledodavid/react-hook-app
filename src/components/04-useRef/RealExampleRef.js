import React, { useState } from 'react';
import '../02-useEffect/effects.css';
import MultipleCustomHooks from '../03-examples/MultipleCustomHooks';


const RealExampleRef = () => {

  const [show, setShow] = useState(false);

  return(
    <div>
      <h1>RealExampleRef</h1>
      <hr/>

      {show && <MultipleCustomHooks />}

      <button className="btn btn-primary mt-4" onClick={() => setShow(!show)}>Show / Hide</button>
      
    </div>
  );
}


export default RealExampleRef;