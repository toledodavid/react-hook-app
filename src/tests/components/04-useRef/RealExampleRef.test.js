import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import RealExampleRef from '../../../components/04-useRef/RealExampleRef';

describe('Tests for <RealExampleRef /> component', () => {

  const wrapper = shallow(<RealExampleRef/>);

  test('It should render component correctly', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
  });

  test('It should <MultipleCustomHooks /> component', () => {
    wrapper.find('button').simulate('click');
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
  });
});