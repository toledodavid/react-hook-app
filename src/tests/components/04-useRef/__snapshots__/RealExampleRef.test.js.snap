// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <RealExampleRef /> component It should render component correctly 1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-4"
    onClick={[Function]}
  >
    Show / Hide
  </button>
</div>
`;
